browser/src/react-redux/App/Signup/SignupForm.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from "react";
+import axios from "axios";
+import Form from "./Form";
+import utility from "./signupUtility";
+
+class signupForm extends Component {
+  constructor() {
+    super();
+    this.state = {
+      email: "",
+      password: "",
+      confirmPassword: "",
+      errors: {
+        confirmPassword: "",
+        password: "",
+        email: "",
+        general: ""
+      },
+      buttonDisabled: false
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  handleChange(key, e) {
+    this.setState({
+      [key]: e
+    });
+  }
+  onSubmit(e) {
+    this.setState({
+      errors: {
+        general: "",
+        email: "",
+        password: ""
+      }
+    });
+
+    e.preventDefault();
+
+    const user = {
+      email: this.state.email,
+      password: this.state.password
+    };
+
+    axios
+      .post("/api/user", user)
+      .then(res => {
+        if (res.request.status === 201) {
+          console.log(`Success!: ${JSON.stringify(res.data.user, null, "  ")}`);
+        }
+      })
+      .catch(err => {
+        console.log(
+          `Error!: ${JSON.stringify(err.response.data.error, null, "  ")}`
+        );
+        if (err.response.data.error.errors) {
+          let newErrors = utility.extractErrors(err.response.data.error.errors);
+
+          this.setState({
+            errors: { ...this.state.errors, ...newErrors }
```

Comment:
double check if need to restore/save any errors that you're _not_ replacing using `setState`